src/toast-notification/categories.scss

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+// See LICENSE.md in the project root for license terms and full copyright notice.
+@import '../style/index';
+@import './toast';
+
+/// category must be one of: positive, negative, primary (informational), warning
+@mixin iui-toast-category($category) {
+  @include themed {
+    border-color: t(iui-color-foreground-#{$category});
```

Comment:
I didn't know that you can have dynamic variable names using `#{$category}`. Good job removing duplicate code!